earn/src/components/borrow/modal/tab/AddUniswapNFTAsCollateralTab.tsx

```diff
@@ -0,0 +1,381 @@
+    chainId: activeChain.id,
+  });
+  const { writeAsync: writeApproveAsync } = useContractWrite({
+    address: UNISWAP_NONFUNGIBLE_POSITION_MANAGER_ADDRESS,
+    abi: erc721ABI,
+    functionName: 'approve',
+    mode: 'recklesslyUnprepared',
+    chainId: activeChain.id,
+  });
+
+  const data = ethers.utils.defaultAbiCoder.encode(
+    ['uint256', 'int24', 'int24', 'int128', 'uint144'],
+    [
+      uniswapNFTPosition[0],
+      uniswapNFTPosition[1].tickLower,
+      uniswapNFTPosition[1].tickUpper,
+      `-${uniswapNFTPosition[1].liquidity.toString()}`,
+      '0',
+    ]
+  );
+  const { config: contractWriteConfig } = usePrepareContractWrite({
+    address: marginAccount.address,
+    abi: MarginAccountABI,
+    functionName: 'modify',
+    args: [ALOE_II_UNISWAP_NFT_MANAGER_ADDRESS, data, [true, true]],
+    enabled: getApprovedData === ALOE_II_UNISWAP_NFT_MANAGER_ADDRESS,
+    chainId: activeChain.id,
+  });
+  if (contractWriteConfig.request) {
+    contractWriteConfig.request.gasLimit = contractWriteConfig.request.gasLimit.mul(GAS_ESTIMATE_WIGGLE_ROOM).div(100);
+  }
+  const {
+    write: contractWrite,
+    data: contractData,
+    isSuccess: contractDidSucceed,
+    isLoading: contractIsLoading,
+  } = useContractWrite(contractWriteConfig);
+
+  useEffect(() => {
+    if (contractDidSucceed && contractData) {
+      setPendingTxn(contractData);
+      setIsPending(false);
+      setIsOpen(false);
+    } else if (!contractIsLoading && !contractDidSucceed) {
+      setIsPending(false);
+    }
+  }, [contractDidSucceed, contractData, contractIsLoading, setPendingTxn, setIsOpen]);
+
+  let confirmButtonState = ConfirmButtonState.READY;
+
+  if (approvingTxn !== null) {
+    confirmButtonState = ConfirmButtonState.APPROVING;
+  } else if (isPending) {
+    confirmButtonState = ConfirmButtonState.PENDING;
+  } else if (getApprovedData !== ALOE_II_UNISWAP_NFT_MANAGER_ADDRESS) {
+    confirmButtonState = ConfirmButtonState.APPROVE_NFT_MANAGER;
+  } else if (contractWriteConfig && contractWriteConfig.request === undefined) {
+    confirmButtonState = ConfirmButtonState.LOADING;
+  }
+
+  const confirmButton = getConfirmButton(confirmButtonState);
+
+  return (
+    <FilledStylizedButton
+      size='M'
+      fillWidth={true}
+      disabled={!confirmButton.enabled}
+      onClick={() => {
+        if (confirmButtonState === ConfirmButtonState.APPROVE_NFT_MANAGER) {
+          setIsPending(true);
+          writeApproveAsync?.({
+            recklesslySetUnpreparedArgs: [
+              ALOE_II_UNISWAP_NFT_MANAGER_ADDRESS,
+              BigNumber.from(uniswapNFTPosition[0].toFixed(0)),
+            ],
+            recklesslySetUnpreparedOverrides: { gasLimit: BigNumber.from(100000) },
+          })
+            .then((txnResult) => {
+              setApprovingTxn(txnResult);
+              txnResult
+                .wait(1)
+                .then(() => {
+                  refetchGetApprovedData();
+                })
```

Comment:
nit: Could be written as `.then(refetchGetApprovedData)`

Reply:
In the future, if you choose not to address a comment, please reply to it instead of just resolving 🙂

Reply:
I did address this, but the linter must have reverted it